photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -138,9 +156,9 @@ class CoordinateDescent(
               .setName(s"Updated model with coordinateId $coordinateId at iteration $iteration")
               .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
               .materialize()
-
           case _ =>
         }
+        // We need to split out the following 2 match expressions: [[FactoredRandomEffectModel]] matches both
         oldModel match {
           case broadcastLike: BroadcastLike => broadcastLike.unpersistBroadcast()
           case _ =>
```

Comment:
I am not that familiar with Scala, but can you please explain why the match statement is done twice? Couldn't you put broadcastLike and rddLike in the same match statement? Sorry if it is a stupid question, just trying to understand here.